client/src/game/ui/websocket.ts

```diff
@@ -598,10 +598,11 @@ commands.set('replayLeader', (data: ReplayLeaderData) => {
   // Hide the replay area if we are in a hypothetical
   if (globals.hypothetical) {
     hypothetical.show();
+  } else {
+    // A batchDraw() already happens in the hypothetical.show() function
```

Comment:
the point of batch draw is that you can call it N times and it will incur no additional cost. (as opposed to the normal draw function which will actually always redraw the layer instantly)

thus its okay to just call batch draw judiciously whenever you need it

Reply:
Oh that's good to know, I won't spend time trying to kill them in the future